src/group_theory/complement.lean

```diff
@@ -0,0 +1,297 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+
+import group_theory.group_action
+import group_theory.order_of_element
+import group_theory.quotient_group
+
+/-!
+# Complements
+
+In this file we define the complement of a subgroup.
+
+## Main definitions
+
+- `is_complement S T` where `S` and `T` are subsets of `G`
+- `left_transversals S`
+- `left_transversals T`
+
+## Main results
+
+- `is_complement_of_coprime` : Subgroups of coprime order are complements
+- `schur_zassenhaus` : Schur-Zassenhaus for abelian normal subgroups
+-/
+
+open_locale big_operators
+
+namespace subgroup
+
+variables {G : Type*} [group G] (H K : subgroup G) (S T : set G)
+
+/-- `S` and `T` are compelements if `(*) : S × T → G` is a bijection -/
+def is_complement : Prop := ∀ g : G, ∃! x : S × T, x.1.1 * x.2.1 = g
```

Comment:
Is there a reason not to `@[to_additive]` everything in this file?

Reply:
`@[to_additive]` mostly works, although it has trouble in a few places